code/cfile/cfile.cpp

```diff
@@ -732,7 +732,7 @@ CFILE* _cfopen(const char* source, int line, const char* file_path, const char*
 
 		FILE *fp = fopen(longname, happy_mode);
 		if (fp)	{
-			return cf_open_fill_cfblock(source, line, fp, dir_type);
+			return cf_open_fill_cfblock(source, line, file_path, fp, dir_type);
```

Comment:
`file_path` can be a full path here rather than just the file name. There is an earlier check to see if it's a full path so you can just grab the filename then if possible, otherwise you would have to split it yourself.

Reply:
Ok.